### PR TITLE
Add inline editor component

### PR DESCRIPTION
**Pull request by xvicmanx**
**Release Type:** *New feature*

Fixes https://x-team-internal.atlassian.net/browse/XP-2157

## Description

These changes add a new Generic Inline Editor Component.

## Checklist

- [x] set yourself as an assignee
- [x] make sure your code lints (`npm run lint`)
- [x] Flow typechecks passed (`npm run typecheck`)
- [x] **manually tested the app** by running it in the browser and checked nothing is broken and operates as expected!

## Related PRs

List related PRs against other Auto repos if applicable:

branch | PR
------ | ------
p-2158-profile-details-module-update-items | [Auto API](https://github.com/x-team/auto-api/pull/492)
xp-2159-allow-editing-fields | [Auto](https://github.com/x-team/auto/pull/1521)

## Screenshots
inline editor component - 2

### CODE REVIEW

**Comment by bernardodiasc** [comment]
hey @xvicmanx for UX perspective I have some suggestions:

1. When performing editions, some UI highlight could be used to demonstrate that have changes in progress. The highlight could be a border color or something like that. Right now it's not visible if have unsaved changes.

screen recording 2018-09-17 at 09 58 am

2. The buttons styles are a bit misleading, because the featured button is currently the "cancel", this should be inverted: left button should be "cancel" and have the gray styles, right button should be "save" with highlighted styles :)

screen shot 2018-09-17 at 10 04 37 am

3. Considering both suggestions above, make sure to disable the "save" button if there's nothing to save.

Let me know any question or concerns :)

File: src/components/ui/InlineEditor.js [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+import ClickOutside from 'react-click-outside'
+
+import Button from './Button'
+
+const cmz = require('cmz')
+
+const cx = {
+  link: cmz(`
+    & {
+      text-decoration: none;
```
[the remark this answers is not in the document]
Comment (xvicmanx) [reply src/components/ui/InlineEditor.js]: Thanks for the suggestion Bernardo. I will removed them.

File: src/components/ui/InlineEditor.js [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+
+type PresenterProps = {
+  value: any,
+  isEditable: boolean,
+  isHover: boolean,
+  activateEditingMode(): void
+}
+
+type Props = {
+  /** Component's value */
```
[the remark this answers is not in the document]
Comment (xvicmanx) [reply src/components/ui/InlineEditor.js]: Thanks :)

**Comment by bernardodiasc** [comment]
Observed another strange behaviour:

4. Edit -> save -> click on text again make lose the previous data

screen recording 2018-09-17 at 10 11 am

I noticed in your screen record that you included the "unsaved changes" state, but for some reason I couldn't not see that in my testing on local.

File: src/components/ui/InlineEditor.js [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+    const props = {
+      value: editValue,
+      onValueChange: this.handleValueChange,
+      activateEditingMode: this.handleActivateEditingMode,
+      isEditable,
+      isHover
+    }
+
+    return (
+      <ClickOutside onClickOutside={this.handleClickOutside}>
```
[the remark this answers is not in the document]
Comment (xvicmanx) [reply src/components/ui/InlineEditor.js]: Great suggestion. I will apply those changes to the InlineEditor as well.

**Comment by xvicmanx** [comment]
Hi @bernardodiasc, I pushed the suggested changes.
However, I was not able to replicate the strange behavior of point 4.

**Comment by xvicmanx** [comment]
Hi @rdparedes, one thing you could do is to wrap the 'Edit mode' component of the `Inline editor` in a container and add margin to it. Something like 
```
.wrapper {
   margin-bottom: 20px
}



editor={(props) => (
  <div className="wrapper">
   <TextArea />
</div>
)}
```

What do you think?

File: src/components/ui/InlineEditor.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (nicksp) [reply src/components/ui/InlineEditor.js]: No reasons to not do that from now on 😄

**Pull request merged**